src/Main.cpp

```diff
@@ -1,10 +1,223 @@
+        usage(cerr);
+        return 1;
+    }
+
+    string cmd = args.front();
+    args.pop_front();
+
+    std::unique_ptr<genset_whisperpower_ddc::VariableSpeedMaster> genset_whisperpower_ddc_master;
+    auto* master = new genset_whisperpower_ddc::VariableSpeedMaster();
+    genset_whisperpower_ddc_master.reset(master);
+    master->openURI(uri);
+
+    if (cmd == "send_control_command") {
+        if (args.size() != 1) {
+            cerr << "wrong number of arguments\n\n";
+            usage(cerr);
+            return 1;
+        }
+
+        string controlCommandArg = args.front();
+        ControlCommand controlCommand;
+
+        if (controlCommandArg == "start") {
+            controlCommand = CONTROL_CMD_START;
+        }
+        else if (controlCommandArg == "stop") {
+            controlCommand = CONTROL_CMD_STOP;
+        }
+        else if (controlCommandArg == "keep_alive") {
+            controlCommand = CONTROL_CMD_KEEP_ALIVE;
+        }
+        else {
+            cerr << "unknown control command\n\n";
+            usage(cerr);
+            return 1;
+        }
+
+        args.pop_front();
+
+        genset_whisperpower_ddc_master->sendControlCommand(
+            controlCommand
+        );
+    }
+    else if (cmd == "read-frame") {
+        if (args.size() != 0) {
+                cerr << "command read-frame does not accept arguments\n\n";
+                usage(cerr);
+                return 1;
+        }
+
+        Frame frame = genset_whisperpower_ddc_master->readFrame();
+
+        if (frame.targetID != 0x0081) {
+            cerr << "unknown target ID '0x" <<  std::hex << (uint16_t) frame.targetID << "'\n\n";
+            return 1;
+        }
+
+        if (frame.sourceID != 0x0088) {
+            cerr << "unknown source ID '0x" <<  std::hex << (uint16_t) frame.sourceID << "'\n\n";
+            return 1;
+        }
+
+        if (frame.command == 2){
+            int rpm = (frame.payload[1] << 8) | frame.payload[0];
+            int udcStartBattery = (frame.payload[3] << 8) | frame.payload[2];
+            std::bitset<8> statusA(frame.payload[4]);
+            bool overallAlarm = statusA[0];
+            bool engineTempAlarm = statusA[1];
+            bool pmVoltageAlarm = statusA[2];
+            bool oilPressAlarm = statusA[3];
+            bool exhaustTempAlarm = statusA[4];
+            bool uac1Alarm = statusA[5];
+            bool iac1Alarm = statusA[6];
+            bool oilPressHighAlarm = statusA[7];
+            std::bitset<8> statusB(frame.payload[5]);
+            bool lowStartBatteryVoltAlarm = statusB[2];
+            bool startFailure = statusB[4];
+            bool runSignal = statusB[5];
+            bool startByOpUnit = statusB[7];
+            std::bitset<8> statusC(frame.payload[6]);
+            bool modelDetection50Hz = statusC[2];
+            bool modelDetection60Hz = statusC[3];
+            bool modelDetection3Phase = statusC[4];
+            bool modelDetectionMobile = statusC[5];
```

Comment:
Remember to get rid of this once you have moved the conversion to the status structure from the component to the driver.